### PR TITLE
operator== and operator!= for QuerySnapshot.

### PR DESCRIPTION
**Pull request by ehsannas**

### CODE REVIEW

**Comment by ehsannas** [comment]
Note that since we're delegating the actual equality check to the underlying implementation, the newly added test does not cover all possible cases of equality/inequality. It mainly serves as a tool to ensure we do invoke the underlying implementation.

File: firestore/src/android/query_snapshot_android.h [hunk 1]
```diff
@@ -65,6 +65,13 @@ class QuerySnapshotInternal : public Wrapper {
   std::size_t size() const;
 };
 
+bool operator==(const QuerySnapshotInternal& lhs,
+                const QuerySnapshotInternal& rhs);
+inline bool operator!=(const QuerySnapshotInternal& lhs,
```
Comment (var-const) [inline firestore/src/android/query_snapshot_android.h:70]: Nit: separate the functions by an empty line?

File: firestore/integration_test_internal/src/query_snapshot_test.cc [hunk 2]
```diff
@@ -30,5 +31,42 @@ TEST_F(QuerySnapshotTest, Assignment) {
 
 #endif  // defined(__ANDROID__)
 
+TEST_F(QuerySnapshotTest, TestCanQueryWithAndWithoutDocumentKey) {
+  CollectionReference collection =
+      Collection({{"a", {{"k", FieldValue::String("a")}}},
+                  {"b", {{"k", FieldValue::String("b")}}},
+                  {"c", {{"k", FieldValue::String("c")}}}});
+  QuerySnapshot snapshot1 = ReadDocuments(collection.Limit(2));
```
Comment (var-const) [inline firestore/integration_test_internal/src/query_snapshot_test.cc:39]: Can you also include a default-constructed `QuerySnapshot` into the comparisons?

**Comment by var-const** [comment]
@ehsannas Oh, and this needs a release note as well.

File: firestore/src/android/query_snapshot_android.h (hunk 1 above)
Comment (ehsannas) [reply firestore/src/android/query_snapshot_android.h:71]: Done

File: firestore/integration_test_internal/src/query_snapshot_test.cc (hunk 2 above)
Comment (ehsannas) [reply firestore/integration_test_internal/src/query_snapshot_test.cc:39]: Done

**Review by var-const — APPROVED** [review]

**Pull request merged**